File: yunu/프로그래머스/덧칠하기/내 코드.js
function solution(n, m, section) {
  let count = 0;
  let start = -1;
  for (let i = 0; i < section.length; i++) {
    if (start === -1) {
      start = section[i] - 1;
      count++;
    } else if (section[i] - start > m) {
      i--;
      start = -1;
    }
  }

  return count;
}

File: yunu/프로그래머스/보석 쇼핑/내 코드.py
from collections import defaultdict

def solution(gems):
    n = len(gems)
    totalType = len(set(gems))
    currType = defaultdict(int)
    interval = []
    
    end = 0
    for start in range(n):
        while len(currType) < totalType and end < n:
            currType[gems[end]] += 1
            end += 1
        
        if len(currType) == totalType:
            interval.append([start + 1, end])
            
        currType[gems[start]] -= 1
        if currType[gems[start]] == 0:
            del currType[gems[start]]
            
    return min(interval, key=lambda inter: inter[1] - inter[0])
